rca/people/models.py

```diff
@@ -1,65 +1,219 @@
+            [InlinePanel("role", label=_("Staff role"))], heading=_("Staff roles")
         ),
+        MultiFieldPanel([FieldPanel("email")], heading=_("Contact information")),
+        FieldPanel("job_title"),
         FieldPanel("introduction"),
+        FieldPanel("body"),
+        MultiFieldPanel(
+            [
+                FieldPanel("research_highlights_title"),
+                InlinePanel(
+                    "related_project_pages", label=_("Project pages"), max_num=8
+                ),
+            ],
+            heading=_("Research highlights gallery"),
+        ),
+        StreamFieldPanel("gallery"),
+        MultiFieldPanel(
+            [
+                FieldPanel("more_information_title"),
+                StreamFieldPanel("more_information"),
+            ],
+            heading=_("More information"),
+        ),
+        StreamFieldPanel("related_links"),
     ]
+
+    edit_handler = TabbedInterface(
+        [
+            ObjectList(content_panels, heading="Content"),
+            ObjectList(key_details_panels, heading="Key details"),
+            ObjectList(BasePage.promote_panels, heading="Promote"),
+            ObjectList(BasePage.settings_panels, heading="Settings"),
+        ]
+    )
+
+    def format_research_highlights(self):
+        """Internal method for formatting related projects to the correct
+        structure for the gallery template
+
+        Returns:
+            List
+        """
+        items = []
+        for page in self.related_project_pages.all():
+            page = page.page.specific
+            meta = None
+            related_school = page.related_school_pages.first()
+            if related_school is not None:
+                meta = related_school.page.title
+
+            items.append(
+                {
+                    "title": page.title,
+                    "link": page.url,
+                    "image": page.hero_image,
+                    "description": page.introduction,
+                    "meta": meta,
+                }
+            )
+        return items
+
+    def get_roles_grouped(self):
+        items = []
+        # First populate a list of all values
+        # E.G [['role title name','programme title'm 'url'], ['role title name','programme title', 'None'], ...]
+        for value in self.role.all():
+            if value.programme:
+                items.append([value.programme.title, value.role, value.programme.url])
+            else:
+                items.append([value.custom_programme, value.role, None])
+
+        # Create a dictionary of values re-using keys so we can group by both
+        # the programmes and the custom programmes.
+        re_grouped = {}
+        for (key, value, link) in items:
+            if key in re_grouped:
+                re_grouped[key]["items"].append(value)
+            else:
+                re_grouped[key] = {"items": [value]}
+            re_grouped[key]["link"] = link
+        return re_grouped.items()
```

Comment:
I couldn't figure out how to write the query for this, so I had to do it by juggling the data around.

Reply:
@kevinhowbrook Just wondering whether we really need this now? Seems like it would make more sense to obey the order imposed by Orderable, so the editor is in control?

Reply:
If we really do need this, I would personally do it at the database level, by annotating the queryset with a new string (the programme page title (if set), or the custom programme), then use that to apply ordering. Here's a really complex example from Wagtail:
https://github.com/wagtail/wagtail/blob/master/wagtail/core/sites.py#L14

This version would just need one `When` clause, plus the 'default'.